src/neuroconv/tools/nwb_helpers.py

```diff
@@ -47,20 +50,35 @@ def get_default_nwbfile_metadata() -> DeepDict:
 
 def make_nwbfile_from_metadata(metadata: dict) -> NWBFile:
     """Make NWBFile from available metadata."""
-    metadata = dict_deep_update(get_default_nwbfile_metadata(), metadata)
+
+    # Validate metadata
+    schema_path = Path(__file__).resolve().parent.parent / "schemas/base_metadata_schema.json"
+    with open(schema_path) as schema_file:
+        base_metadata_schema = json.load(schema_file)
+    try:
+        # jsonschema.validate(metadata, base_metadata_schema)
+        validate_metadata(metadata=metadata, schema=base_metadata_schema)
+    except jsonschema.exceptions.ValidationError as e:
+        print(f"Base metadata is not valid.\n")
+        raise e
+
     nwbfile_kwargs = metadata["NWBFile"]
+    # convert ISO 8601 string to datetime
+    if isinstance(nwbfile_kwargs.get("session_start_time"), str):
+        nwbfile_kwargs["session_start_time"] = datetime.fromisoformat(nwbfile_kwargs["session_start_time"])
+    if "session_description" not in nwbfile_kwargs:
+        nwbfile_kwargs["session_description"] = "No description."
```

Comment:
Now this one is interesting, especially insofar as it interacts with the base metadata schema (which is also now being used in the forms by the GUIDE)

Just to clarify...

a) Does the NWB Schema require this field at the top most level for all NWB files?

b) As I recall, the PyNWB `NWBFile` initializer does require it, regardless of (a), can you confirm?

c) But in NeuroConv, we view this field as optional, and so we fill it with a placeholder. Shouldn't we then include a `default` value in the NeuroConv metadata schema itself? Or, we don't want the GUIDE forms to auto-populate placeholders... Even though our forms then do not mark it as a true requirement even though it's needed to ultimately create the file in PyNWB...

Reply:
Out of the 3 NWB required fields (session_description, identifier and session_start_time) we only really enforce `sessions_start_time`. I see the inconsistency you’re pointing out. I agree with keep operating like this. 
Neuroconv (and the schema/guide) aims at facilitating conversion to NWB, and one important aspect of it is auto-filling as much as possible. We auto-fill session_start_time when it’s present on input files, but the other two fields are not possible to auto-extract like that. I guess one could argue for the importance of user-filled `session_description`, but I personally don’t see it for `identifier`, machines are better at doing unique identifiers.
In that sense, neuroconv schema could deviate from nwb schema in specific ways which would improve the user experience, imo.

Now, if we’re auto-filling required fields, we should do this explicitly. IMO, the schema should have a default value AND be required, and that should be reflected in the Forms: placeholder values and a red asterisk +  validation on form submission.